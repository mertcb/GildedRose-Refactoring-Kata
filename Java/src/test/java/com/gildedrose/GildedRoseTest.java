package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void foo() {
        CombinationApprovals.verifyAllCombinations(
            this::test_item,
            new String[]{"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
            new Integer[]{-1, 5, 0, 1,13},
            new Integer[]{0, 20, 49, 10, 75}
        );
    }

    private String test_item(String name, int sellIn, int quality){
        Item[] items = new Item[] { new Item(name, sellIn, quality) };

        GildedRose app = new GildedRose(items);
        app.updateQuality();

        return app.items[0].toString();
    }

}
