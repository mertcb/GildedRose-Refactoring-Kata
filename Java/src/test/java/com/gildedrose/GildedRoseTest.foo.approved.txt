[foo, -1, 0] => foo, -2, 0 
[foo, -1, 20] => foo, -2, 18 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 10] => foo, -2, 8 
[foo, -1, 75] => foo, -2, 73 
[foo, 5, 0] => foo, 4, 0 
[foo, 5, 20] => foo, 4, 19 
[foo, 5, 49] => foo, 4, 48 
[foo, 5, 10] => foo, 4, 9 
[foo, 5, 75] => foo, 4, 74 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 20] => foo, -1, 18 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 10] => foo, -1, 8 
[foo, 0, 75] => foo, -1, 73 
[foo, 1, 0] => foo, 0, 0 
[foo, 1, 20] => foo, 0, 19 
[foo, 1, 49] => foo, 0, 48 
[foo, 1, 10] => foo, 0, 9 
[foo, 1, 75] => foo, 0, 74 
[foo, 13, 0] => foo, 12, 0 
[foo, 13, 20] => foo, 12, 19 
[foo, 13, 49] => foo, 12, 48 
[foo, 13, 10] => foo, 12, 9 
[foo, 13, 75] => foo, 12, 74 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 20] => Aged Brie, -2, 22 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 75] => Aged Brie, -2, 75 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 20] => Aged Brie, 4, 21 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 10] => Aged Brie, 4, 11 
[Aged Brie, 5, 75] => Aged Brie, 4, 75 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 20] => Aged Brie, -1, 22 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 75] => Aged Brie, -1, 75 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 20] => Aged Brie, 0, 21 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 10] => Aged Brie, 0, 11 
[Aged Brie, 1, 75] => Aged Brie, 0, 75 
[Aged Brie, 13, 0] => Aged Brie, 12, 1 
[Aged Brie, 13, 20] => Aged Brie, 12, 21 
[Aged Brie, 13, 49] => Aged Brie, 12, 50 
[Aged Brie, 13, 10] => Aged Brie, 12, 11 
[Aged Brie, 13, 75] => Aged Brie, 12, 75 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 20] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 75] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 20] => Backstage passes to a TAFKAL80ETC concert, 4, 23 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 10] => Backstage passes to a TAFKAL80ETC concert, 4, 13 
[Backstage passes to a TAFKAL80ETC concert, 5, 75] => Backstage passes to a TAFKAL80ETC concert, 4, 75 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 20] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 75] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 20] => Backstage passes to a TAFKAL80ETC concert, 0, 23 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 10] => Backstage passes to a TAFKAL80ETC concert, 0, 13 
[Backstage passes to a TAFKAL80ETC concert, 1, 75] => Backstage passes to a TAFKAL80ETC concert, 0, 75 
[Backstage passes to a TAFKAL80ETC concert, 13, 0] => Backstage passes to a TAFKAL80ETC concert, 12, 1 
[Backstage passes to a TAFKAL80ETC concert, 13, 20] => Backstage passes to a TAFKAL80ETC concert, 12, 21 
[Backstage passes to a TAFKAL80ETC concert, 13, 49] => Backstage passes to a TAFKAL80ETC concert, 12, 50 
[Backstage passes to a TAFKAL80ETC concert, 13, 10] => Backstage passes to a TAFKAL80ETC concert, 12, 11 
[Backstage passes to a TAFKAL80ETC concert, 13, 75] => Backstage passes to a TAFKAL80ETC concert, 12, 75 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 20] => Sulfuras, Hand of Ragnaros, -1, 20 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 75] => Sulfuras, Hand of Ragnaros, -1, 75 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 20] => Sulfuras, Hand of Ragnaros, 5, 20 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, 5, 10] => Sulfuras, Hand of Ragnaros, 5, 10 
[Sulfuras, Hand of Ragnaros, 5, 75] => Sulfuras, Hand of Ragnaros, 5, 75 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 20] => Sulfuras, Hand of Ragnaros, 0, 20 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 75] => Sulfuras, Hand of Ragnaros, 0, 75 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 20] => Sulfuras, Hand of Ragnaros, 1, 20 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 1, 10] => Sulfuras, Hand of Ragnaros, 1, 10 
[Sulfuras, Hand of Ragnaros, 1, 75] => Sulfuras, Hand of Ragnaros, 1, 75 
[Sulfuras, Hand of Ragnaros, 13, 0] => Sulfuras, Hand of Ragnaros, 13, 0 
[Sulfuras, Hand of Ragnaros, 13, 20] => Sulfuras, Hand of Ragnaros, 13, 20 
[Sulfuras, Hand of Ragnaros, 13, 49] => Sulfuras, Hand of Ragnaros, 13, 49 
[Sulfuras, Hand of Ragnaros, 13, 10] => Sulfuras, Hand of Ragnaros, 13, 10 
[Sulfuras, Hand of Ragnaros, 13, 75] => Sulfuras, Hand of Ragnaros, 13, 75 
